src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -298,11 +301,11 @@ public int write(ByteBuffer src) throws IOException {
                 if (!isOpen())
                     return 0;
                 do {
-                    long comp = Blocker.begin();
+                    boolean attempted = Blocker.begin(sync | direct);
```

Comment:
Is the use of the bitwise operator `|` instead of `||` on `boolean` fields, here and other places, intentional?

Reply:
Well spotted, these should be ||.